server/service/src/invoice/inbound_return/insert/validate.rs

```diff
@@ -14,10 +11,9 @@ pub fn validate(
     input: &InsertInboundReturn,
 ) -> Result<Name, InsertInboundReturnError> {
     use InsertInboundReturnError::*;
-    check_invoice_does_not_exists(&input.id, connection).map_err(|e| match e {
-        InvoiceAlreadyExistsError::InvoiceAlreadyExists => InvoiceAlreadyExists,
-        InvoiceAlreadyExistsError::RepositoryError(err) => DatabaseError(err),
-    })?;
+    if let Some(_) = check_invoice_exists(&input.id, connection)? {
```

Comment:
Got rid of check_invoice_does_not_exist method and just reused this one to return error if it is Some

Reply:
Nice!